dig.go

```diff
@@ -119,6 +143,15 @@ func (c *Container) Provide(constructor interface{}, opts ...ProvideOption) erro
 // The function may return an error to indicate failure. The error will be
 // returned to the caller as-is.
 func (c *Container) Invoke(function interface{}, opts ...InvokeOption) error {
+	err := c.invoke(function, opts...)
+	// If error wrapping wasn't requested, return the root cause.
+	if !c.wrapErrors {
+		err = errRootCause(err)
```

Comment:
`func errCause(bool wrap, err)` might be an easier abstraction so you can just pass the `c.WrapErrors` in as opposed to having to write an if statement in each applicable case

Reply:
Moot now.